lib/IRGen/IRGenSIL.cpp

```diff
@@ -2288,8 +2288,13 @@ void IRGenSILFunction::visitFunctionRefBaseInst(FunctionRefBaseInst *i) {
       fn, NotForDefinition, false /*isDynamicallyReplaceableImplementation*/,
       isa<PreviousDynamicFunctionRefInst>(i));
   llvm::Value *value;
+//  auto isSpecialAsyncWithoutCtxtSize =
+//      fn->isAsync() && fn->getName().equals("swift_task_future_wait");
   auto isSpecialAsyncWithoutCtxtSize =
-      fn->isAsync() && fn->getName().equals("swift_task_future_wait");
+      fn->isAsync() && (
+          fn->getName().equals("swift_task_future_wait") ||
+          fn->getName().equals("swift_task_group_poll") // TODO: do we need this?
```

Comment:
TODO: not actively used today; but it may be nicer to call directly into `swift_task_group_poll` so we don't pollute the `swift_task_future_wait` impl with the special case for if it's a group child.